src/radical/pilot/agent/lm/jsrun.py

```diff
@@ -27,23 +27,28 @@ def _configure(self):
 
     # --------------------------------------------------------------------------
     #
-    def _create_resource_set_file(self, slots, uid, sandbox):
+    def _create_resource_set_file(self, slots, uid, sandbox, mpi=False,
+                                  gpu=False):
```

Comment:
I don't understand why we pass `gpu` and `mpi` flags - what difference does it make?  If GPUs are defined in the slots, they need to go into the resource sets - the flag won't change that.  It seems like `mpi` triggers the specification of `rank` - but as discussed, this has not much to do with the MPI ranks, and I'd argue we should use the same syntax for MPI and non-MPI variants (specifying `rank` for non-MPI is fine).

Reply:
I am perfectly fine with that, but:
```bash
bash-4.2$ cat unit.000000.rs 
cpu_index_using: physical
rank: 0: { host: 1; cpu: {0-13}}
rank: 0: { host: 1; cpu: {22-36}}
bash-4.2$ jsrun --erf_input=$pwd/unit.000000.rs /bin/date
ERF error: 1+ ranks are assigned with multiple resource sets.
```
```bash
bash-4.2$ cat unit.000000.rs 
cpu_index_using: physical
1: { host: 1; cpu: {0-13}}
1: { host: 1; cpu: {22-36}}
bash-4.2$ jsrun --erf_input=$pwd/unit.000000.rs /bin/date
Thu Feb 28 10:56:27 EST 2019
Thu Feb 28 10:56:27 EST 2019
```
```bash
bash-4.2$ cat unit.000000.rs 
cpu_index_using: physical
rank: 0: { host: 1; cpu: {0-13}}
bash-4.2$ jsrun --erf_input=$pwd/unit.000000.rs /bin/date
Thu Feb 28 10:57:13 EST 2019
```
```bash
bash-4.2$ cat unit.000000.rs 
cpu_index_using: physical
1: { host: 1; cpu: {0-13}}
bash-4.2$ jsrun --erf_input=$pwd/unit.000000.rs /bin/date
Thu Feb 28 10:57:56 EST 2019
```

In my opinion, for non-MPI unit the ERF file should look like examples 2 and 4, despite the fact that  3 works. I will dot the rank part, and I see how to remove the flags.

Reply:
Isn't 2 an example for an MPI task? Do you mean you require 'rank' only if it an MPI task? The two seems to be conflicting, no?

Reply:
Each line is an application execution. It executes the application defined in the `jsrun` multiple times with as many processes as the number defined at the beginning. I am not sure it will work correctly with MPI

Reply:
Yes, that's precisely how it should work though. I will launch two processes that both execute /bin/date. 
I think my point is connected with my comment below. I'll add an example there. My suggestion is to use
either rank for both MPI and non-MPI applications or not use at rank at all.

Reply:
@iparask Following yesterday discussion on Slack, did we all converge on this?